### DISCUSSION TITLE
What metadata is ImageMagick able to read?

### DISCUSSION THREAD

**Discussion by halloleo** [ImageMagick/ImageMagick#5019] [Help] [answered]
I would like to check which metadata IM reads via it CLI tools reads an image. In particular I need the **Image Title** field and the **Image Description** field.

What tool do I need for this? I tried `identify`, but could  doesn't seem to read either.

### Background:

I am using `Wand`, the Python Wrapper for IM, to programmatically read some image metadata - and to my surprise Wand seems to read only the Image Description, not the Image Title - that's why I'm curious what IM itself can do.

**Comment by snibgo** [comment]
What version of IM?

Which image format?

IM is a raster image processor. When an image file contains metadata that is not relevant to processing an image, IM mostly ignores that metadata.

If your need is processing of metadata (rather than actual images), IM is not the best tool. Better are exiftool and exiv2.

**Reply by halloleo** [reply]
Thanks fro chiming in.

> What version of IM?

6.9 (linux) and 7.1 (macOS)

> Which image format?

JPEGs and TIFFs

> IM is a raster image processor. When an image file contains metadata that is not relevant to processing an image, IM mostly ignores that metadata.

Well, Wand has a `metadata` property which gives back _some_ (EXIF) metadata .- From my peek in Wands source they do come from the Magick API.

> If your need is processing of metadata (rather than actual images), IM is not the best tool. Better are exiftool and exiv2.

I know, I'm a big fan of exiftool. But I would like to keep ther ewuiremnst for my script small. I use Wand for the Image processing and I was hoping I can get Image Description and Image Title (the only two metadata I need) from that as well...

**Comment by fmw42** [comment, marked as answer]
You can try

`identify -verbose image.suffix`

**Reply by halloleo** [reply]
I tried that and got it all: I can see **Image Name** (=Title) and **Caption** (= Image Description).

Now, I only have to figure out why Wand shows me only the Caption / Description. Both Metadata are in the same "profile", so I would think Wnad should list both...

**Reply by fmw42** [reply]
Post your wand code

**Reply by halloleo** [reply]
Sure:

    with Image(in_img_path) as src:
        for (k,v) in src.metadata.items():
            print(f"{k} = '{v}'")

Here I get the Caption through the key `exif:ImageDescription`, but I do not get through any key the Image Name.

On the other hand `identify -verbose in_img_path` gives me the Caption on the output line with `Caption` before the colon and the Image name or Title on line with `Image Name` before the colon.

I guess that's an problem with Wand... That's why I placed a question on their discussion board: [How can I see the image Name/Title in `Image.metadata`? · Discussion #570 · emcconville/wand](https://github.com/emcconville/wand/discussions/570)

**Reply by halloleo** [reply]
Update: The issue might be that Wand only enumerates EXIF metadata, not IPTC metadata.

**Reply by fmw42** [reply]
If you think you have an issue with Wand, then report to the Wand developer as you did. He is the expert. ImageMagick does not maintain Wand.

**Reply by halloleo** [reply]
Yep, makes sense.

**Comment by snibgo** [comment]
For IPTC, see https://imagemagick.org/script/escape.php and the source code MagickCore\identify.c.

If you were using MagickWand, I suspect function GetImageProperty() or similar is what you want. "Wand-py" seems to be a Python interface to MagickWand, and I know nothing about it.

**Reply by halloleo** [reply]
Thanks @snibgo. 

Indeed the Escape page does list it all. I had first overlooked (and then misunderstood) the section about direct access to IPTC data.

Now it's all good: I can access the relevant metadata via `%[IPTC:2:05]` and `%[IPTC:2:120]`.

Thanks again!
